decoder/vaapidecoder_vp9.cpp

```diff
@@ -90,23 +91,50 @@ void VaapiDecoderVP9::flush(void)
 
 Decode_Status VaapiDecoderVP9::ensureContext(const Vp9FrameHdr* hdr)
 {
+    bool resetContext = false;
+    Decode_Status status = DECODE_SUCCESS;
+    DEBUG("got frame size: %d x %d", hdr->width, hdr->height);
+
     // only reset va context when there is a larger frame
     if (m_configBuffer.width < hdr->width
         || m_configBuffer.height <  hdr->height) {
+        resetContext = true;
         INFO("frame size changed, reconfig codec. orig size %d x %d, new size: %d x %d",
                 m_configBuffer.width, m_configBuffer.height, hdr->width, hdr->height);
-        Decode_Status status = VaapiDecoderBase::terminateVA();
+        
+        if (m_hasContext){
+            status = VaapiDecoderBase::terminateVA();
```

Comment:
Guangxin had sent me another email before on this. As per his email - VaapiDecoderBase::terminateVA(void) is not the same as vaTerminate. VaapiDecoderBase::terminateVA(void) will clear the surface pool in VaapiDecoderBase and reset the context. So Guangxin has a separate patch that will remove reset of the external display in the terminateVA. Then we will not lose display handle on resolution change. 

Guangxin, do you want to submit your patch separately to address Daniel's concern? What is your comment on this? Thanks.